src/aspire/utils/rotation.py

```diff
@@ -12,39 +12,18 @@
 
 
 class Rotation:
-    def __init__(self, num_rots, seed=None, matrices=None, dtype=np.float32):
+    def __init__(self, matrices):
         """
          Initialize a Rotation object
 
-         :param num_rots: Total number of rotation sets
-         :param seed: Seed for initializing random rotations.
-             If None, the random generator is not re-seeded.
         :param matrices: Rotation matrices to initialize Rotation object.
-             If None, matrices will be generated from uniformly
-             distributed angles.
-         :param dtype: data type for rotation matrices
         """
-        self.dtype = np.dtype(dtype)
-        self.num_rots = num_rots
-        self._matrices = None
+        self.dtype = matrices.dtype
+        self.num_rots = matrices.shape[0]
```

Comment:
`num_rots` and `shape` are no longer necessary since these can be read out from `_matrices`.

Reply:
Removed in new commit ecfd5fd.